tests/operator/test_operator.cpp

```diff
@@ -56,8 +56,9 @@ TEST(OperatorTest, CheckExpectationValue) {
             generate_random_observable_with_eigen(n, random);
 
         auto state = StateVector::Haar_random_state(n);
+        auto state_cp = state.amplitudes();
         Eigen::VectorXcd test_state = Eigen::VectorXcd::Zero(dim);
-        for (UINT i = 0; i < dim; ++i) test_state[i] = state[i];
+        for (UINT i = 0; i < dim; ++i) test_state[i] = state_cp[i];
```

Comment:
CPU から直接 GPU 上のデータにアクセスできないため、vector へのコピーを挟みます

Reply:
これ、出来なさげでは有るんですが手元テストが通っているので本当にできないのか不信です。どこかにドキュメントがありますか。

Reply:
私の環境ではやはりこうしないとエラーが出ますね...。
```

Reply:
ドキュメントで言及されている箇所は見つからないのですが、私の環境(WSL2+Docker)ではやはりエラーが出ます...。以下で再現できませんか？
```cpp
#include <iostream>

#include "state/state_vector.hpp"

using namespace qulacs;

int main() {
    Kokkos::initialize();
    auto state = StateVector::Haar_random_state(3);
    state[2];
    Kokkos::finalize();
}
```
```
$ ./build/exe/main
Kokkos::OpenMP::initialize WARNING: OMP_PROC_BIND environment variable not set
  In general, for best performance with OpenMP 4.0 or better set OMP_PROC_BIND=spread and OMP_PLACES=threads
  For best performance with OpenMP 3.1 set OMP_PROC_BIND=true
  For unit testing set OMP_PROC_BIND=false

Kokkos::Cuda::initialize WARNING: running kernels compiled for compute capability 6.1 on device with compute capability 8.6 , this will likely reduce potential performance.
Kokkos::View ERROR: attempt to access inaccessible memory space (label="state")
Backtrace:
[0x55b4cc319d59] 
[0x55b4cc2ee4a0] 
[0x55b4cc2da97a] 
[0x55b4cc2c4edd] 
[0x55b4cc2bea21] 
[0x7f8cfb2edd90] 
[0x7f8cfb2ede40] __libc_start_main
[0x55b4cc2c0b75] 
Aborted
```

Reply:
言及されている箇所を見つけました。
以下のページでDLできるPDFの49枚目のスライドに失敗例として紹介されています。
https://github.com/kokkos/kokkos-tutorials/blob/main/LectureSeries/KokkosTutorial_02_ViewsAndSpaces.pdf

Reply:
了解です。調査ありがとうございます。